### PR TITLE
[docs] Update REST API docs

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Fix typos and improve content
- Add /autocomplete APIs
- Add /describe APIs

### CODE REVIEW

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ Then a JWT token is returned and needs to be passed as a bearer in the headers f
 
 **Wrong credentials**: there is currently no error on bad authentication but instead a 302 redirect to the login page, e.g.:
 
-    curl -X POST https://demo.gethue.com/notebook/execute/v1/create_notebook -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjIxNjM5NjMxLCJqdGkiOiI0NTY3NTA4MzM5YjY0MjFmYTMzZDJjMzViZWUyMDAyMCIsInVzZXJfaWQiOjF9.qrMNrr69eo38dOsV2aYp8k6WqBeyJZkbSuavxA_o_kM"
+    curl -X POST https://demo.gethue.com/notebook/execute/query/create_notebook -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjIxNjM5NjMxLCJqdGkiOiI0NTY3NTA4MzM5YjY0MjFmYTMzZDJjMzViZWUyMDAyMCIsInVzZXJfaWQiOjF9.qrMNrr69eo38dOsV2aYp8k6WqBeyJZkbSuavxA_o_kM"
 
     [21/May/2021 16:26:46 -0700] middleware   INFO     Redirecting to login page: /notebook/execute/v1/create_notebook
     [21/May/2021 16:26:46 -0700] access       INFO     127.0.0.1 -anon- - "POST /notebook/execute/v1/create_notebook HTTP/1.1" - (mem: 172mb)-- login redirection
```
Comment (Harshg999) [inline docs/docs-site/content/developer/api/rest/_index.md:119]: Trying to run this current API gives 404 error
When trying to run /api/query/create_notebook/ with wrong JWT token gives 401 unauthorized access (token invalid or expired) and not 302 redirect.

Do we need to update this section too @romainr ?

**Comment by romainr** [comment]
This is the wrong api path, it should now start with /api/...

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 2]
```diff
@@ -113,7 +113,7 @@ Then a JWT token is returned and needs to be passed as a bearer in the headers f
 
 **Wrong credentials**: there is currently no error on bad authentication but instead a 302 redirect to the login page, e.g.:
 
-    curl -X POST https://demo.gethue.com/notebook/execute/v1/create_notebook -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjIxNjM5NjMxLCJqdGkiOiI0NTY3NTA4MzM5YjY0MjFmYTMzZDJjMzViZWUyMDAyMCIsInVzZXJfaWQiOjF9.qrMNrr69eo38dOsV2aYp8k6WqBeyJZkbSuavxA_o_kM"
+    curl -X POST https://demo.gethue.com/notebook/execute/query/create_notebook -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjIxNjM5NjMxLCJqdGkiOiI0NTY3NTA4MzM5YjY0MjFmYTMzZDJjMzViZWUyMDAyMCIsInVzZXJfaWQiOjF9.qrMNrr69eo38dOsV2aYp8k6WqBeyJZkbSuavxA_o_kM"
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:116]: /api/editor/... not /notebook

**Review by romainr — APPROVED** [review]

File: docs/docs-site/content/developer/api/rest/_index.md (hunk 2 above)
Comment (Harshg999) [reply docs/docs-site/content/developer/api/rest/_index.md]: Fixed

**Pull request merged**